File: backend/src/domain/user/Staff.ts
import { Clinic } from "@/domain/clinic";
import {StaffRole} from "./StaffRole";
import {User} from "./User";

export class Staff extends User {
  constructor(id?: string,
              firstname?: string,
              lastname?: string,
              patronymic?: string,
              birthday?: Date,
              avatarUrl?: string,
              phone?: string,
              email?: string,
              public clinic?: Clinic,
              public role?: StaffRole) {
    super(id,
    firstname,
    lastname,
    patronymic,
    birthday,
    avatarUrl,
    phone,
    email);
  }
}

File: backend/src/main/java/com/hypocrates/hypocrates/domain/Staff.java
package com.hypocrates.hypocrates.domain;

import lombok.*;

import java.util.Date;

@Getter
public class Staff extends User {
    private Clinic clinic;

    private StaffRole role;

    public Staff(Long id, String firstname, String lastname, String patronymic, Date birthday, String avatarUrl, String email, String phone, Boolean emailIsActive, Boolean phoneIsActive, Clinic clinic, StaffRole role, String password) {
        super(id, firstname, lastname, patronymic, birthday, avatarUrl, email, phone, emailIsActive, phoneIsActive, password);
        this.clinic = clinic;
        this.role = role;
    }

}
